app/src/main/java/fr/free/nrw/commons/upload/UploadActivity.java

```diff
@@ -426,6 +426,14 @@ public boolean isWLMUpload() {
 
     private void receiveExternalSharedItems() {
         uploadableFiles = contributionController.handleExternalImagesPicked(this, getIntent());
+
+        try {
+            Collections.sort(uploadableFiles, (f1, f2) -> (int) (
+                f2.getFileCreatedDate(this).getEpochDate() -
+                    f1.getFileCreatedDate(this).getEpochDate()));
+        }catch (NullPointerException e){
+            e.printStackTrace();
```

Comment:
Please use the logging framework, as seen in many other places in the app. Thanks!

Reply:
Hi!, used `Timber` on my latest commit, i hope that was right